contrib/antlr4/src/test/java/org/apache/accumulo/access/grammar/antlr/AccessExpressionAntlrEvaluator.java

```diff
@@ -0,0 +1,121 @@
+
+    @Override
+    public String toString() {
+      return "Entity [authorizations=" + authorizations + "]";
+    }
+
+  }
+
+  private final List<Entity> entities;
+
+  public AccessExpressionAntlrEvaluator(List<Authorizations> authSets) {
+    entities = new ArrayList<>(authSets.size());
+
+    for (Authorizations a : authSets) {
+      Set<String> entityAuths = a.asSet();
+      Entity e = new Entity();
+      entities.add(e);
+      e.authorizations = new HashSet<>(entityAuths.size() * 2);
+      a.asSet().stream().forEach(auth -> {
+        e.authorizations.add(auth);
+        String quoted = AccessExpression.quote(auth);
+        if (!quoted.startsWith("\"")) {
+          quoted = '"' + quoted + '"';
+        }
+        e.authorizations.add(quoted);
+      });
+    }
+    // System.out.println("AUTHS: " + entities);
+  }
+
+  public boolean canAccess(byte[] accessExpression) throws IllegalAccessExpressionException {
+    return canAccess(AccessExpression.of(accessExpression));
+  }
+
+  public boolean canAccess(AccessExpression accessExpression) {
+    return canAccess(accessExpression.getExpression());
+  }
+
+  public boolean canAccess(String accessExpression) {
+    if ("".equals(accessExpression)) {
+      return true;
+    }
+    return canAccess(AccessExpressionAntlrParser.parseAccessExpression(accessExpression));
+  }
+
+  public boolean canAccess(Access_expressionContext parsedExpression) {
+    for (Entity e : entities) {
+      if (!evaluate(e, parsedExpression)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private Boolean evaluate(Entity e, ParseTree node) {
+    Boolean retval;
+    if (node instanceof Access_tokenContext) {
+      retval = e.authorizations.contains(node.getText());
+    } else if (node instanceof TerminalNode
+        || node instanceof And_operatorContext | node instanceof Or_operatorContext) {
+      retval = null;
+    } else {
+      int childCount = node.getChildCount();
+      int trueCount = 0;
+      int falseCount = 0;
+      for (int i = 0; i < childCount; i++) {
+        Boolean childResult = evaluate(e, node.getChild(i));
+        if (childResult == null) {
+          continue;
+        } else {
+          if (childResult == Boolean.TRUE) {
+            trueCount++;
+          } else {
+            falseCount++;
+          }
+        }
+      }
+      if (childCount == 1) {
+        retval = falseCount == 0;
+      } else {
+        if (node instanceof Or_expressionContext) {
+          retval = trueCount > 0;
+        } else if (node instanceof And_expressionContext) {
+          retval = trueCount > 0 && falseCount == 0;
+        } else {
+          retval = trueCount > 0 && falseCount == 0;
+        }
+      }
+      // System.out.println("child results-> true=" + trueCount +", false=" + falseCount);
+    }
+    // System.out.println("node: " + node.getClass().getSimpleName()+ ", value: " + node.getText() +
+    // ", retval: " + retval);
```

Comment:
If these were added for debugging and are no longer needed, could remove.  If they may still be useful please ignore this comment.

```suggestion
```